src/metrics.jl

```diff
@@ -394,6 +394,20 @@ weuclidean(a, b, w) = WeightedEuclidean(w)(a, b)
     s3 = min(s2, p - s2)
     abs2(s3)
 end
+
+function (pe::PeriodicEuclidean{<:AbstractMatrix})(a::AbstractVector, b::AbstractVector)
+    p = parameters(pe)
+    ndim = length(a)
+    @assert size(p,1) == size(p, 2) == length(b) == ndim  # check dimension
```

Comment:
More in line with the surrounding code (the multiple dispatch would work on the fact the parameter `p` is a matrix:
```suggestion
function _evaluate(dist::PeriodicEuclidean, a::AbstractVector, b::AbstractVector, p::AbstractMatrix)
    ndim = LinearAlgebra.checksquare(p)
    @boundscheck if length(a) != length(b)
        throw(DimensionMismatch("first array has length $(length(a)) which does not match the length of the second, $(length(b))."))
    end
    @boundscheck if length(a) != ndim
        throw(DimensionMismatch("arrays have length $(length(a)) but basis vectors have length $ndim."))
    end
```
And maybe move the function a little to not disrupt the sequence of `eval_*` functions?